core/utils.py

```
@@ -280,3 +283,137 @@ def track_email_sent(email_address: str, email_type: EmailType, profile: Profile
+
+        file_output = output[0] if isinstance(output, list) else output
+        image_url = str(file_output)
+
+        logger.info(
+            "[GenerateOGImage] Image generated successfully",
+            blog_post_id=generated_post.id,
+            project_id=generated_post.project_id,
+            image_url=image_url,
+        )
+
+        image_response = urlopen(image_url)
```

Ruff (0.14.3)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)